File: AppUtils/tags/V00-03-00/src/AppDataPath.cpp
//--------------------------------------------------------------------------
// File and Version Information:
// 	$Id$
//
// Description:
//	Class AppDataPath...
//
// Author List:
//      Andy Salnikov
//
//------------------------------------------------------------------------

//-----------------------
// This Class's Header --
//-----------------------
#include "AppUtils/AppDataPath.h"

//-----------------
// C/C++ Headers --
//-----------------
#include <stdlib.h>
#include <boost/algorithm/string.hpp>
#include <boost/filesystem.hpp>

//-------------------------------
// Collaborating Class Headers --
//-------------------------------

//-----------------------------------------------------------------------
// Local Macros, Typedefs, Structures, Unions and Forward Declarations --
//-----------------------------------------------------------------------

namespace fs = boost::filesystem;

//		----------------------------------------
// 		-- Public Function Member Definitions --
//		----------------------------------------

namespace AppUtils {

//----------------
// Constructors --
//----------------
AppDataPath::AppDataPath(const std::string& relPath)
  : m_path()
{
  // get $SIT_DATA
  const char* dataPath = getenv("SIT_DATA");
  if (not dataPath) return;

  // split SIT_DATA path on :
  std::list<std::string> paths;
  boost::split(paths, dataPath, boost::is_any_of(":"));

  // find first existing file
  for (std::list<std::string>::const_iterator it = paths.begin(); it != paths.end(); ++ it) {

    fs::path path = *it;
    path /= relPath;

    if (fs::exists(path)) {
      m_path = path.string();
      break;
    }

  }
}

} // namespace AppUtils

File: CalibManager/trunk/src/AppDataPath.py
#--------------------------------------------------------------------------
# File and Version Information:
#  $Id$
#
# Description:
#  Module AppDataPath...
#
#------------------------------------------------------------------------

"""AppDataPath class represents a path to a file that can be found in
one of the $SIT_DATA locations.

@version $Id$

@author Andy Salnikov
"""

#------------------------------
#  Module's version from SVN --
#------------------------------
__version__ = "$Revision$"
# $Source$

#--------------------------------
#  Imports of standard modules --
#--------------------------------
import sys
import os

#---------------------------------
#  Imports of base class module --
#---------------------------------

#-----------------------------
# Imports for other modules --
#-----------------------------

#---------------------
#  Class definition --
#---------------------
class AppDataPath ( object ) :
    """
    AppDataPath class represents a path to a file that can be found in one of the $SIT_DATA locations.
    """

    #----------------
    #  Constructor --
    #----------------
    def __init__ (self, relPath) :
        """Constructor takes relative file path."""
        self.m_path = ""
        
        dataPath = '../../data:' + os.getenv("SIT_DATA")
        if not dataPath: return
            
        for dir in dataPath.split(':'):
            path = os.path.join(dir, relPath)
            if os.path.exists(path):
                self.m_path = path
                break
            
    #-------------------
    #  Public methods --
    #-------------------

    def path(self) :
        """Returns path of the existing file or empty string"""
        return self.m_path

#
#  In case someone decides to run this module
#
if __name__ == "__main__" :

    # In principle we can try to run test suite for this module,
    # have to think about it later. Right now just abort.
    sys.exit ( "Module is not supposed to be run as main module" )
